streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -662,6 +664,16 @@ object StreamingContext extends Logging {
     }
   }
 
+  /**
+   * Get the context that is currently active, that is, started but not stopped.
+   */
+  @Experimental
+  def getActive(): Option[StreamingContext] = {
+    ACTIVATION_LOCK.synchronized {
+      Option(activeContext.get())
```

Comment:
Took another look. Yes, the lock seems good to me. 

> What if I call start() then stop() currently on the same streamingContext in different threads.

@pwendell streamingContext.start and stop are `synchronized`.